docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+                output_mode='int',
+                output_sequence_length=seq_length)       
+        # Make a text-only dataset (without labels), then call adapt
+            train_text = raw_train_ds.map(lambda x, y: x)
+            vectorize_layer.adapt(train_text)
+        except Exception as e:
+            logger.error(f"Error while attempting to create vectorize layer: \n {str(e)}")
+            raise e
+        logger.info("Vectorization Layer complete...")
+        try:
+            def vectorize_text(text, label):
+                text = tf.expand_dims(text, -1)
+                return vectorize_layer(text), label
+
+            train_ds = raw_train_ds.map(vectorize_text)
+            val_ds = raw_val_ds.map(vectorize_text)
+            test_ds = raw_test_ds.map(vectorize_text)
+            logger.info("Dataset mapped to text")
+            AUTOTUNE = tf.data.AUTOTUNE
+            train_ds = train_ds.cache().prefetch(buffer_size=AUTOTUNE)
+            test_ds = test_ds.cache().prefetch(buffer_size=AUTOTUNE)           
+            val_ds = val_ds.cache().prefetch(buffer_size=AUTOTUNE)
+            logger.info("Dataset cached")
+            model = tf.keras.Sequential([
+                layers.Embedding(max_features + 1, embedding_dim),
+                layers.Dropout(0.2),
+                layers.GlobalAveragePooling1D(),
+                layers.Dropout(0.2),
+                layers.Dense(1)])
+            model.compile(loss=losses.BinaryCrossentropy(from_logits=True),
+                optimizer='adam',
+                metrics=tf.metrics.BinaryAccuracy(threshold=0.0))
+            logger.info("beginning training...")
+            history = model.fit(train_ds, validation_data=val_ds,
+                    epochs=epochs)
+            mp_queue.put(history)
+            logger.info("Saving...")
+            model.save(model_save_loc, save_format='tf', overwrite=True)
+        except Exception as e:
+            logger.error(f"In precompile {e}")
+
+        try:
+            logger.info(f"Training complete!")
+            export_model = tf.keras.Sequential([
+                vectorize_layer,
+                model,
+                layers.Activation('sigmoid')
+                ])
+            logger.info(f"Compiling new model...")
+            export_model.compile(
+                loss=losses.BinaryCrossentropy(from_logits=False), optimizer="adam", metrics=['accuracy']
+            )
+            loss, accuracy = model.evaluate(test_ds)
+            logger.info(f"Evaluation Results: \n {str(loss)} \n {str(accuracy)}")
+            export_model.save(f"{model_save_loc}-exported", save_format="tf", overwrite=True)
+        except Exception as e:
+            logger.error(f"ERR in TRAIING \n {e}")
+            raise e
+```
+
+While this method may be intimidating, don't let it scare you! It is simply the core code of our companion piece, gently refactored to fit into our tool's (and the PythonSdk's) runtime and lifecycle.
```

Comment:
TW review:

While this method might be intimidating, don't let it scare you! It's simply the core code of the companion piece, gently refactored to fit into the tool's (and the Python SDK's) runtime and lifecycle.